src/main/todos.ts

```diff
@@ -59,4 +88,153 @@ export function getBasecampTodoRequest(content: string, description: string, ass
         notify: notify,
         due_on: basecampDueDate
     }
+}
+
+/**
+ * Sends requests to create new Basecamp Todos
+ * 
+ * @param basecampRequests array of BasecampTodoRequest objects to send
+ * @returns a map that associates role titles with basecamp todo ids
+ */
+export function createNewTodos(roleRequestMap: RoleRequestMap): RoleTodoIdMap {
+    const roleTodoIdMap: RoleTodoIdMap = {};
+
+    Object.keys(roleRequestMap).forEach( role => {
+        let request = roleRequestMap[role];
+        let basecampTodoId: string = createTodo(request, DEFAULT_TODOLIST_IDENTIFIER)
+        roleTodoIdMap[role] = basecampTodoId;
+    });
+
+    return roleTodoIdMap;
+}
+
+/**
+ * Deletes the list of todos based on their todo ids
+ * 
+ * @param todoIds todo ids corresponding to todos to delete
+ */
+export function deleteTodos(todoIds: string[]): void {
+
+    for(const id of todoIds) {
+
+        let todoIdentifier: TodoIdentifier = {
+            projectId: PROJECT_ID,
+            todoId: id
+        }
+
+        deleteTodo(todoIdentifier);
+    }
+}
+
+/**
+ * Deletes todos associated with helper groups that are no longer present on the event row.
+ * The first line checks whether there are any rows present in the original roles that are not present in the current roles (indicating an obsolete role)
+ * 
+ * @param roleRequestMap a map associating an event's roles with api request bodies
+ * @param currentRoleTodoIdMap a map associating an event's roles to existing todo ids, that is currently saved in the document properties
+ * @returns a string array of obsolete roles that were deleted
+ */
+export function deleteObsoleteTodos(roleRequestMap: RoleRequestMap, currentRoleTodoIdMap: RoleTodoIdMap): string[] {
+    Logger.log("Checking for removed roles...\n")
+    const removedRoles: string[] = getRoles(roleRequestMap, currentRoleTodoIdMap, RoleStatus.REMOVED);
+
+    if(removedRoles.length > 0) {
+        const obsoleteTodoIds: string[] = getObsoleteTodosIds(removedRoles, currentRoleTodoIdMap);
+        deleteTodos(obsoleteTodoIds);
+
+    } else {
+        Logger.log("No removed roles detected!\n");
+    }
+
+    return removedRoles;
+}
+
+/**
+ * Gets a list of obsolete todo ids based off of the obsolete roels roles by comparing the original and current roles together
+ * 
+ * @param obsoleteRoles an array of obsolete roles
+ * @param roleTodoIdMap a map associating an event's roles with todo ids
+ * @returns an array of obsolete roles
+ */
+export function getObsoleteTodosIds(obsoleteRoles: string[], roleTodoIdMap: RoleTodoIdMap): string[] {
```

Comment:
Because the role status of `obselete` was renamed, functions and variables like this should be renamed accordingly

Reply:
I think it's ok. Thanks for the clarification